### PR TITLE
Updates button to not always render longestText

### PR DESCRIPTION
**Pull request by kierangillen**
- `longestText` is no longer rendered unless the prop is passed to the `Button` component

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit d0dd1ec158da6427265ac98c91f4b8dc519e7282

https://deploy-preview-548--artsy-palette.netlify.com

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 21b1d35b0afa79134ae49fead8d3fc2553c1b749

https://deploy-preview-548--artsy-palette.netlify.com

File: packages/palette/src/elements/Button/Button.tsx [hunk 1]
```diff
@@ -184,22 +184,31 @@ export class ButtonBase extends Component<ButtonBaseProps & SansProps> {
...
+            >
+              {longestText}
+            </HiddenText>
+          </>
+        )}
+        {!longestText && (
+          <Sans pt="1px" weight={weight || "medium"} size={size}>
+            {children}
+          </Sans>
+        )}
```
Comment (damassi) [inline packages/palette/src/elements/Button/Button.tsx]: maybe 
```js
{longestText 
  ? ...
  : ...
}
```

Comment (kierangillen) [reply packages/palette/src/elements/Button/Button.tsx]: Yes, just switched to the ternary, it is cleaner.

**Comment by artsyit** [comment]
:rocket: PR was released in v4.14.38 :rocket:

**Pull request merged**